symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/annotation/SlashAnnotationProcessor.java

```diff
@@ -0,0 +1,168 @@
+
+import com.symphony.bdk.core.activity.ActivityRegistry;
+import com.symphony.bdk.core.activity.command.CommandContext;
+
+import lombok.Generated;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.aop.framework.autoproxy.AutoProxyUtils;
+import org.springframework.aop.scope.ScopedObject;
+import org.springframework.aop.scope.ScopedProxyUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.MethodIntrospector;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.core.annotation.AnnotationUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * Registers {@link Slash} methods as individual {@link com.symphony.bdk.core.activity.command.SlashCommand} instances
+ * within the {@link ActivityRegistry}.
+ *
+ * @see <a href="https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-factory-extension-bpp">BeanPostProcessor</a>
+ * @see <a href="https://github.com/spring-projects/spring-framework/blob/master/spring-context/src/main/java/org/springframework/context/event/EventListenerMethodProcessor.java">EventListenerMethodProcessor.java</a>
+ */
+@Slf4j
+public class SlashAnnotationProcessor implements BeanPostProcessor, ApplicationContextAware {
+
+  /** The list of packages to be ignored from application context scanning */
+  private static final String[] IGNORED_PACKAGES = {
+      "org.springframework.",
+      "com.symphony.bdk.gen.",
+      "com.symphony.bdk.core."
+  };
+
+  /** The {@link ActivityRegistry} is used here to register the slash activities */
+  private final ActivityRegistry registry;
+
+  private ConfigurableApplicationContext applicationContext;
+
+  public SlashAnnotationProcessor(ActivityRegistry registry) {
+    this.registry = registry;
+  }
+
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    this.applicationContext = (ConfigurableApplicationContext) applicationContext;
+  }
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+
+    if (!ScopedProxyUtils.isScopedTarget(beanName)) {
+
+      final Class<?> type = this.determineTargetClass(beanName);
+
+      if (type != null) {
+        try {
+          this.processBean(bean, beanName, type);
+        }
+        catch (Throwable ex) {
+          // just alert the developer
+          log.warn("Failed to process @Slash annotation on bean with name '" + beanName + "'", ex);
```

Comment:
{} placeholders are usually used (for performance mainly here it won't be a problem but to be consistent)